### PR TITLE
[mlir][vector] Use `DenseI64ArrayAttr` in vector.multi_reduction

### PR DESCRIPTION
**Pull request by MacDue**
This prevents some unnecessary conversions to/from int64_t and IntegerAttr.

### CODE REVIEW

**Comment by MacDue** [comment]
This is the final one of these tidy ups (alongside #101850)

File: mlir/lib/Dialect/Vector/IR/VectorOps.cpp [hunk 1]
```diff
@@ -467,8 +466,8 @@ LogicalResult MultiDimReductionOp::verify() {
   Type inferredReturnType;
   auto sourceScalableDims = getSourceVectorType().getScalableDims();
   for (auto it : llvm::enumerate(getSourceVectorType().getShape()))
-    if (!llvm::any_of(getReductionDims().getValue(), [&](Attribute attr) {
-          return llvm::cast<IntegerAttr>(attr).getValue() == it.index();
+    if (!llvm::any_of(getReductionDims(), [&](int64_t dim) {
+          return dim == static_cast<int64_t>(it.index());
         })) {
       targetShape.push_back(it.value());
       scalableDims.push_back(sourceScalableDims[it.index()]);
```
Comment (kuhar) [inline mlir/lib/Dialect/Vector/IR/VectorOps.cpp:473]: nit: we could use structured bindings for index and value here. This is existing code, so feel free to ignore.

**Review by kuhar — APPROVED** [review]
Nice

**Review by joker-eph — APPROVED** [review]
Thanks!

**Pull request merged**